src/Workspaces/Remote/Core/AbstractAssetProvider.cs

```diff
@@ -103,14 +103,14 @@ async Task<ImmutableArray<DocumentInfo>> CreateDocumentInfosAsync(ChecksumsAndId
     }
 
     public async Task<ImmutableArray<T>> CreateCollectionAsync<T>(
```

Comment:
>CreateCollectionAsync [](http://example.com/codeflow?start=41&length=21)

Can you clarify why GetAssetsAsync isn't virtual on AbstractAssetProvider and used here? Seems like bulk synchronization would be nice to do here.

Reply:
seems reasonable.  i'll make that change.

Reply:
ChecksumSynchronizer.CollectChecksumChildrenAsync seems similar

Reply:
updated to have that pattern.

Reply:
changed that one as well.